### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by omarsalem7**
## What does this PR do?

This PR is raised to add Person, Student and Teacher classes

## Description of the task to be completed

- [x] Create class `Person`
- [x] Create class `Student`
- [x] Create class `Teacher`
- [x] Inherit `Student` and `Teacher` from `Person` class

## How should it be tested

Clone the repository
```
git clone https://github.com/omarsalem7/OOP-school-library.git
```
Navigate to the folder
```
cd OOP-school-library
```
Checkout the branch
```
git checkout milestone1
```

#### Thanks for reviewing my PR 😊

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  end
+
+  def can_use_services?
+    @parent_permission || is_of_age?
+  end
+
+  private
+
+  def of_age?
+    @age > 18
```
Comment (rustam-tolipov) [inline person.rb]: - [x] Age must be greater or equal to the 18. Kindly fix this issue.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,11 @@
 ![](https://img.shields.io/badge/Microverse-blueviolet)
 
-# Project Title
-
-> Description of the project
```
Comment (rustam-tolipov) [inline README.md:5]: - [x] [OPTIONAL] I would highly suggest you personalize your MIT license. As you can see in this image I highlighted places that need to be changed or removed.

Screen Shot 2022-04-11 at 7 46 28 PM
.

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(classroom, age, name, parent_permission)
```
Comment (rustam-tolipov) [inline student.rb]: - [x] [OPTIONAL] I would suggest you set parent_permission to true. As an example:
>   def initialize(age, classroom, name = "Unknown", parent_permission = true)

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @omarsalem7, 

Good job so far!

### HIGHLIGHTS!

- No linters errors.💯

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (omarsalem7) [reply person.rb]: Hello @Rustamxon7 😊
Thanks for your feedback,
I did the changes which you stated, you can check them now

**Review by micheaol — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hi @omarsalem7 , 

Great work on making the changes requested by the previous reviewer 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 
- You set up linter files and no errors. 💯 

well-1

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**